Czaple/AppDelegate.swift

```diff
@@ -10,60 +10,23 @@ import UIKit
 
 @UIApplicationMain
 class AppDelegate: UIResponder, UIApplicationDelegate {
-
+    
+    private let firstLaunch = FirstLaunch()
     var window: UIWindow?
 
-
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        // Override point for customization after application launch.
-        
-        window = UIWindow()
-        window?.makeKeyAndVisible()
-        if UserDefaults.standard.object(forKey: "firstTime") == nil {
-            UserDefaults.standard.set(true, forKey: "firstTime")
-        }
-        UserDefaults.standard.synchronize()
-        
-        if UserDefaults.standard.bool(forKey: "firstTime") == true {
-            let layout = UICollectionViewFlowLayout()
-            layout.scrollDirection = .horizontal
-            let introViewController = IntroSwipingViewController(collectionViewLayout: layout)
-            UserDefaults.standard.set(false, forKey: "firstTime")
-            UserDefaults.standard.synchronize()
-            
-            window?.rootViewController = introViewController
-        } else {
-            let mainViewController = MainView()
-            let navigationController = UINavigationController(rootViewController: mainViewController)
-            window?.rootViewController = navigationController
-        }
         
+        let attractionProvider = AttractionProvider()
+        let boardProvider = BoardProvider()
+        let boardAudioPlayer = BoardAudioPlayer()
+        let boardAlert = BoardAlert(boardProvider: boardProvider, boardAudioPlayer: boardAudioPlayer)
         
+        let introProvider = IntroProvider(attractionProvider: attractionProvider, boardAlert: boardAlert)
+        window = UIWindow()
+        guard let window = window else { return true }
+        let firstTime = firstLaunch.checkFirstLaunch()
+        introProvider.showIntro(window: window, first: firstTime)
```

Comment:
Intro provider should provide viewConttroller to be shown rather you should pass window instance to it. :) 
```swift 
window.rootViewController = itroProvider.viewController(first: firstTime)
```